src/main/scala/com/fulcrumgenomics/bam/CallConsensusFromUmis.scala

```diff
@@ -0,0 +1,118 @@
+   @arg(flag="2", doc="The Phred-scaled error rate for an error post the UMIs have been integrated.") val errorRatePostUmi: PhredValue = DefaultErrorRatePostUmi,
+   @arg(flag="q", doc="Cap the maximum base quality in the input.") val maxBaseQuality: PhredValue = DefaultMaxBaseQuality,
+   @arg(flag="s", doc="Subtract this base quality from the input base qualities.") val baseQualityShift: PhredValue = DefaultBaseQualityShift,
+   @arg(flag="N", doc="Mask (make 'N') consensus bases with quality less than this threshold.") val minConsensusBaseQuality: PhredValue = DefaultMinConsensusBaseQuality,
+   @arg(flag="M", doc="The minimum number of reads to produce a consensus base.") val minReads: Int = DefaultMinReads,
+   @arg(flag="Q", doc="The minimum mean base quality across a consensus base to output.") val minMeanConsensusBaseQuality: PhredValue = DefaultMinMeanConsensusBaseQuality
+  ) extends FgBioTool with LazyLogging {
+
+  Io.assertReadable(input)
+  Seq(output, rejects).foreach(Io.assertCanWriteFile(_))
+  if (attribute.length != 2) throw new ValidationException("attribute must be of length 2")
+  if (errorRatePreUmi < 0)   throw new ValidationException("Error rate pre UMI must be >= 0")
+  if (errorRatePostUmi < 0)  throw new ValidationException("Error rate post UMI must be >= 0")
+
+  /** TODO
+    * - what about when a read has an indel causing the rest of the bases to mismatch?
+    * - what about if a read is soft-clipped and the rest of the bases mismatch (ex. 30M120S) [second pass]?
+    *
+    * TODO: hard clipping?
+    *
+    */
+
+  /** Main method that does the work of reading input files, creating the consensus reads, and writing the output file. */
+  override def execute(): Unit = {
+    val in  = SamReaderFactory.make().open(input.toFile)
+    val out = new SAMFileWriterFactory().makeWriter(in.getFileHeader, true, output.toFile, null)
+    val rej = new SAMFileWriterFactory().makeWriter(in.getFileHeader, true, rejects.toFile, null)
+    // TODO: metrics...
+
+    val options = new ConsensusCallerOptions(
+      attribute                   = attribute,
+      errorRatePreUmi             = errorRatePreUmi,
+      errorRatePostUmi            = errorRatePostUmi,
+      maxBaseQuality              = maxBaseQuality,
+      baseQualityShift            = baseQualityShift,
+      minConsensusBaseQuality     = minConsensusBaseQuality,
+      minReads                    = minReads,
+      minMeanConsensusBaseQuality = minMeanConsensusBaseQuality
+    )
+
+    val progress = new ProgressLogger(logger)
+    val consensusCaller = new ConsensusCaller(
+      input          = in.iterator().asScala,
+      header         = in.getFileHeader,
+      readNamePrefix = readNamePrefix,
+      readGroup      = readGroup,
+      options        = options,
+      rejects        = Some(rej),
+      progress       = Some(progress)
+    )
+
+    consensusCaller.foreach { rec => out.addAlignment(rec) }
```

Comment:
Why bother passing the progress logger into the caller - you could just do it here?